crates/openvino/src/core.rs

```diff
@@ -40,6 +47,155 @@ impl Core {
+            devices.push(device_type);
+        }
+
+        unsafe { ov_available_devices_free(std::ptr::addr_of_mut!(ov_available_devices)) };
+        Ok(devices)
+    }
+
+    /// Gets properties related to this Core.
+    /// The method extracts information that can be set via the [set_property] method.
+    pub fn property(&self, key: PropertyKey) -> Result<Cow<str>> {
+        let ov_prop_key = cstr!(key.as_ref());
+        let mut ov_prop_value = std::ptr::null_mut();
+        try_unsafe!(ov_core_get_property(
+            self.ptr,
+            EMPTY_C_STR.as_ptr(),
+            ov_prop_key,
+            std::ptr::addr_of_mut!(ov_prop_value)
+        ))?;
+        let rust_prop = unsafe { CStr::from_ptr(ov_prop_value) }.to_string_lossy();
+        Ok(rust_prop)
+    }
+
+    /// Sets a property for this Core instance.
+    pub fn set_property(&mut self, key: RwPropertyKey, value: &str) -> Result<()> {
+        let ov_prop_key = cstr!(key.as_ref());
+        let ov_prop_value = cstr!(value);
+        try_unsafe!(ov_core_set_property(
+            self.ptr,
+            EMPTY_C_STR.as_ptr(),
+            ov_prop_key,
+            ov_prop_value,
+        ))?;
+        Ok(())
+    }
+
+    /// Sets properties for this Core instance.
+    pub fn set_properties<'a>(
+        &mut self,
+        properties: impl IntoIterator<Item = (RwPropertyKey, &'a str)>,
+    ) -> Result<()> {
+        for (prop_key, prop_value) in properties {
+            self.set_property(prop_key, prop_value)?;
+        }
+        Ok(())
+    }
+
+    /// Gets properties related to device behaviour.
+    /// The method extracts information that can be set via the [set_device_property] method.
+    pub fn device_property(
+        &self,
+        device_name: impl AsRef<str>,
+        key: PropertyKey,
+    ) -> Result<Cow<str>> {
+        let ov_device_name = cstr!(device_name.as_ref());
+        let ov_prop_key = cstr!(key.as_ref());
+        let mut ov_prop_value = std::ptr::null_mut();
+        try_unsafe!(ov_core_get_property(
+            self.ptr,
+            ov_device_name,
+            ov_prop_key,
+            std::ptr::addr_of_mut!(ov_prop_value)
+        ))?;
+        let rust_prop = unsafe { CStr::from_ptr(ov_prop_value) }.to_string_lossy();
+        Ok(rust_prop)
+    }
+
+    /// Sets a property for a device.
+    pub fn set_device_property(
+        &mut self,
+        device_name: impl AsRef<str>,
+        key: RwPropertyKey,
+        value: &str,
+    ) -> Result<()> {
+        let ov_device_name = cstr!(device_name.as_ref());
+        let ov_prop_key = cstr!(key.as_ref());
+        let ov_prop_value = cstr!(value);
+        try_unsafe!(ov_core_set_property(
+            self.ptr,
+            ov_device_name,
+            ov_prop_key,
+            ov_prop_value,
+        ))?;
+        Ok(())
+    }
+
+    /// Sets properties for a device.
+    pub fn set_device_properties<'a>(
```

Comment:
Is the `'a` still needed?

Reply:
Yes I get an error otherwise, `anonymous lifetimes in impl Trait are unstable`